test.js

```diff
@@ -65,23 +65,25 @@ test('match exact URLs', t => {
 });
 
 test('match URLs in text', t => {
-	const fixture = [
-		'Lorem ipsum //dolor.sit',
-		'<a href="http://example.com">example.com</a>',
-		'[and another](https://another.example.com)',
-		'Foo //bar.net/?q=Query with spaces'
-	].join('\n');
+	const fixture = `
+		Lorem ipsum //dolor.sit
+		<a href="http://example.com">example.com</a>
+		<a href="http://example.com/with-path">with path</a>
+		[and another](https://another.example.com) and
+		Foo //bar.net/?q=Query with spaces
+	`;
 
 	const expected = [
 		'//dolor.sit',
 		'http://example.com',
+		'http://example.com/with-path',
 		'https://another.example.com',
 		'//bar.net/?q=Query'
 	];
 
 	const actual = fixture.match(fn());
 
-	expected.forEach((url, i) => t.is(actual[i], url));
+	t.same(actual, expected);
```

Comment:
Could we keep the old assertion logic? Comparing two long lists in the output is much harder on humans than just seeing the particular URLs that failed to match individually.